worker/runtime/cni_network.go

```diff
@@ -303,6 +333,46 @@ func (n cniNetwork) generateResolvConfContents() ([]byte, error) {
 	return []byte(contents), err
 }
 
+func (n cniNetwork) restrictHostAccess() error {
+	err := n.ipt.CreateChainOrFlushIfExists(filterTable, "INPUT")
+	if err != nil {
+		return fmt.Errorf("create chain or flush if exists failed: %w", err)
+	}
+
+	hostIp, err := GetHostIp()
+	if err != nil {
+		return err
+	}
+	n.ipt.AppendRule(filterTable, "INPUT", "-i", n.config.BridgeName, "-d", hostIp, "-j", "DROP")
```

Comment:
Is there a reason for using `DROP` rather than `REJECT`? Seems like the main difference is that with `DROP`, trying to reach the host will timeout (rather than with `REJECT`, which will fail ASAP)

Reply:
`REJECT` explicitly responds to the attacker and lets them know that this IP exists. `DROP` prevents leaking that information.

Reply:
That brings up another question, actually - does this rule only `DROP` packets destined for the IP of the `eth0` (or equivalent) interface? Will it prevent access to any IP that's accessible within the host's network?

e.g. I was investigating a customer issue where a VM's BOSH DNS server was inaccessible to containers, which was causing DNS issues. This DNS server had IP `169.254.0.2` (which wasn't the same as `eth0` on the host). Would this rule handle that case?

Reply:
Yup. It won't affect any other IPs. We tested this by bringing up a second worker and were able to ping that worker from the container. Guardian has the same behaviour.

Initially, we were wondering if we should try to stop that too but we concluded VM to VM network interactions in the network are up to the operator to determine through their firewall rules (they can technically prevent it through `CONCOURSE_CONTAINERD_RESTRICTED_NETWORK` as well).

Reply:
Not quite sure why the DNS issue came up in the customer issue. Possible it was their internal firewall rules or some nuance about Guardian (if they're using that runtime).

Reply:
Sorry, should've clarified - it was a result of allow host access (i.e. with host access disabled, they couldn't access the internal DNS server, and could access it when enabled).

I guess the question I have is what's the reason for the different behaviour from Guardian? My interpretation of Guardian's `iptables` rules here are:
1. If packets are destined for the external network interface (eth0), then accept them
2. If packets are destined for any other interface, then reject if host access is disabled, and accept if enabled

...which seems like it should work more generally, as it seems local services can bind to other IP addresses (not sure how it works, but e.g. BOSH DNS binding to 169.254.0.2). Let me know if my interpretation is flawed, not 100% sure about how `iptables`/linux networking works evidently